packages/ctl/ctl.js

```diff
@@ -300,6 +395,117 @@ instanceConfigCommands.add(new libCommand.Command({
+		yargs.positional("editor", { describe: "Editor to use",
+			type: "string",
+			default: "" });
+	}],
+	handler: async function(args, control) {
+		let instanceId = await libCommand.resolveInstance(control, args.instance);
+		let response = await libLink.messages.getInstanceConfig.send(control, { instance_id: instanceId });
+		let tmpFile = await libFileOps.getTempFile("ctl-", "-tmp", os.tmpdir());
+		let editor = 0;
+		if (args.editor) {
+			editor = args.editor;
+			/* eslint-disable */
+		} else if (process.env.EDITOR) {
+			editor = process.env.EDITOR;
+		} else if (process.env.VISUAL) {
+			editor = process.env.VISUAL
+			/* eslint-enable */
+			// needed for the process.env statements to not be flagged by eslint
+			// prio for editors is CLI arg > env.EDITOR > env.VISUAL
+		} else {
+			throw new libErrors.CommandError("No editor avalible. Checked CLI input, EDITOR and VISUAL env vars");
+		}
+
+		let allConfigElements = "";
+		let disallowedList = {"instance.id": 0, "instance.assigned_slave": 0, "factorio.settings": 0};
+		for (let group of response.serialized_config.groups) {
+			for (let [name, value] of Object.entries(group.fields)) {
+				if (`${group.name}.${name}` in disallowedList) {
+					continue;
+				}
+				let desc = "";
+				try {
+					desc += libConfig.MasterConfig.groups.get(group.name)._definitions.get(name).description;
+				} catch (err) {
+					desc += "No description found";
+				}
+				allConfigElements += `# ${desc}\n`;
+				// split onto two lines for readability and es-lint
+				if (String(value) === "null") {
+					value = "";
+				}
+				allConfigElements += `${group.name}.${name} = ${value}\n\n`;
+			}
+		}
+		// this could be shorter but the lines would have to be very long, so imo this is better for readability
+		await fs.writeFile(tmpFile, allConfigElements, (err) => {
+			if (err) {
+				throw err;
+			}
+		});
+		let editorSpawn = child_process.spawn(editor, [tmpFile], {
+			stdio: "inherit",
+  			detached: false,
+		});
+		editorSpawn.on("data", (data) => {
+  			process.stdout.pipe(data);
+		});
+		let emmiter = new events.EventEmitter();
+		editorSpawn.on("exit", async (exit) => {
+			const data = await fs.readFile(tmpFile, "utf8");
+			let splitData = data.split(/\r?\n/);
+			// split on newlines
+			let filtered = splitData.filter((value) => !(value[0] === "#")).filter((a) => a);
+			// the last filter removes empty elements left by the first. Not done on one line due to readability.
+			for (let index in filtered) {
+				if (index in filtered) {
+					filtered[index] = filtered[index].split("=");
+					let finalIndex = filtered[index][0].trim();
+					// split on the = we added earlier
+					let part = "";
+					try {
+						part = filtered[index][1].trim();
+						// it's a string if we can read it
+					} catch (err) {
+						// if we can't read it, it's a empty field and therefor null
+						part = "";
+					}
```

Comment:
This part of the editor spawning is the same as for the master. Could it be refactored into a shared parser function and a split message portion?